odoo/models.py

```diff
@@ -5379,22 +5380,32 @@ def convert_pgerror_unique(model, fields, info, e):
     if not ufields or model._table != table:
         return {'message': tools.ustr(e)}
 
+    del cr  # otherwise _() would get the closed cursor from above
     # TODO: add stuff from e.diag.message_hint? provides details about the constraint & duplication values but may be localized...
     if len(ufields) == 1:
         field_name = ufields[0]
-        field = fields[field_name]
-        message = _(u"The value for the field '%s' already exists (this is probably '%s' in the current model).") % (field_name, field['string'])
+        field_description = _convert_field_name(model, field_name)
```

Comment:
Not sure about the function's name, explicitly denoting that this is about translation would probably be a good idea.